### ISSUE TITLE
Trick makes values of variables visible

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#466]
v2.33.20-7-g59a2
```
?- [user]. [(([A]-->[B])-->[(A,B)])]. end_of_file.
   true.

?- (([1]-->[2])-->[X]).
   X = (_A,_B). % unexpected

?- _=_, (([1]-->[2])-->[X]).  % trick trigger _=_
   X = (1,2).
```

**Comment by infradig** [comment]
$ tpl
?- [user]. [(([A]-->[B])-->[(A,B)])]. end_of_file.
   true.
?- (([1]-->[2])-->[X]).
   X = (_A,_B).
?-

You can also trick it with trace...

$ tpl -t
?- [user]. [(([A]-->[B])-->[(A,B)])]. end_of_file.
   true.
?- (([1]-->[2])-->[X]).
[user:1:cp0] CALL true
[user:2:cp0] EXIT true
[user:3:cp0] CALL ([1]-->[2])-->[_0]
[user:4:cp0] EXIT ([1]-->[2])-->[(1,2)]
   X = (1,2).
?-

**Comment by flexoron** [comment]
Yes, but I see CALL true, EXIT true and this does not work:
```
$ tpl
?- [user]. [(([A]-->[B])-->[(A,B)])]. end_of_file.
   true.

?- true, (([1]-->[2])-->[X]).
   X = (_A,_B).

The interesting thing is an independent unification ( _=_ ) which fiddled the displaying.

?- _=_, (([1]-->[2])-->[X]).
   X = (1,2).
```

**Comment by infradig** [comment]
It's the printing that fixes it, some init issue...

$ tpl
?- [user]. [(([A]-->[B])-->[(A,B)])]. end_of_file.
   true.
?- write(ok), nl, (([1]-->[2])-->[X]).
ok
   X = (1,2).

**Comment by flexoron** [comment]
T(r)icketed!

**Comment by flexoron** [comment]
Ticket(=Issue) fixed with v2.33.21-8-g62ce
